ethcore/src/keys_directory.rs

```diff
@@ -0,0 +1,1093 @@
+// Copyright 2015, 2016 Ethcore (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Keys Directory
+
+use common::*;
+use std::path::{PathBuf};
+
+const CURRENT_DECLARED_VERSION: u64 = 3;
+const MAX_KEY_FILE_LEN: u64 = 1024 * 80;
+const MAX_CACHE_USAGE_TRACK: usize = 128;
+
+/// Cipher type (currently only aes-128-ctr)
+#[derive(PartialEq, Debug, Clone)]
+pub enum CryptoCipherType {
+	/// aes-128-ctr with 128-bit initialisation vector(iv)
+	Aes128Ctr(U128)
+}
+
+#[derive(PartialEq, Debug, Clone)]
+enum KeyFileVersion {
+	V3(u64)
+}
+
+/// key generator function
+#[derive(PartialEq, Debug, Clone)]
+pub enum Pbkdf2CryptoFunction {
+	/// keyed-hash generator (HMAC-256)
+	HMacSha256
+}
+
+#[derive(Clone)]
+#[allow(non_snake_case)]
+/// Kdf of type `Pbkdf2`
+/// https://en.wikipedia.org/wiki/PBKDF2
+pub struct KdfPbkdf2Params {
+	/// desired length of the derived key, in octets
+	pub dkLen: u32,
```

Comment:
it's also called `dklen` in the same wikipedia article. see the equation that reads:
```
DK = T1 || T2 || ... || Tdklen/hlen
```

i think in any case rust language conventions should take precedence.